packages/engine/src/scene/systems/SceneDeltaState.tsx

```diff
@@ -0,0 +1,83 @@
+specific language governing rights and limitations under the License.
+
+The Original Code is Infinite Reality Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Infinite Reality Engine team.
+
+All portions of the code written by the Infinite Reality Engine team are Copyright © 2021-2023 
+Infinite Reality Engine. All Rights Reserved.
+*/
+
+import { Component, defineSystem, Entity, getComponent, hasComponent, SerializedComponentType } from '@ir-engine/ecs'
+import { NodeID, NodeIDComponent } from '@ir-engine/engine/src/gltf/NodeIDComponent'
+import { SourceComponent, SourceID } from '@ir-engine/engine/src/scene/components/SourceComponent'
+import { defineState, getMutableState, NO_PROXY_STEALTH, useHookstate, useMutableState } from '@ir-engine/hyperflux'
+import { useEffect } from 'react'
+import { SceneState } from '../../gltf/GLTFState'
+import { SceneObjectSystem } from './SceneObjectSystem'
+
+export type SceneDeltaEntry<C extends Component> = Record<string, Partial<SerializedComponentType<C>>>
+
+export type MaterialDeltaEntry = Record<'materialParameters', any>
+
+export type SceneDeltaRegistry = Record<SourceID, Record<NodeID, SceneDeltaEntry<any> | MaterialDeltaEntry>>
+
+export const SceneDeltaState = defineState({
+  name: 'SceneDeltaState',
+  initial: {} as SceneDeltaRegistry,
+  registerDelta<C extends Component>(entity: Entity, component: C, delta: Partial<SerializedComponentType<C>>) {
+    if (!hasComponent(entity, SourceComponent) || !hasComponent(entity, NodeIDComponent)) return
+    if (!component.jsonID) return
+    const sourceID = getComponent(entity, SourceComponent).replaceAll(/\?hash=[^-]+/g, '')
+    const nodeID = getComponent(entity, NodeIDComponent)
+    const state = getMutableState(SceneDeltaState)
+    if (!state.value[sourceID]) state[sourceID].set({})
+    const source = state[sourceID]
+    if (!source.value[nodeID]) source[nodeID].set({} as SceneDeltaEntry<C>)
+    const componentMap = source[nodeID].get(NO_PROXY_STEALTH) as SceneDeltaEntry<C>
+    componentMap[component.jsonID] = delta
+  },
+  registerMaterialDelta(entity: Entity, props: any) {
+    if (!hasComponent(entity, SourceComponent) || !hasComponent(entity, NodeIDComponent)) return
+    const sourceID = getComponent(entity, SourceComponent).replaceAll(/\?hash=[^-]+/g, '')
+    const nodeID = getComponent(entity, NodeIDComponent)
+    const state = getMutableState(SceneDeltaState)
+    if (!state.value[sourceID]) state[sourceID].set({})
+    const source = state[sourceID]
+    if (!source.value[nodeID]) source[nodeID].set({} as MaterialDeltaEntry)
+    const componentMap = source[nodeID].get(NO_PROXY_STEALTH) as MaterialDeltaEntry
+    componentMap['materialParameters'] = props
+  }
+})
+
+export const SceneDeltaSystem = defineSystem({
+  uuid: 'ir.engine.SceneDeltaSystem',
+  insert: { with: SceneObjectSystem },
+  reactor: () => {
```

Comment:
if this system is just used to define a reactor, that reactor can be moved to `SceneDeltaState` and we can get rid of the system

Reply:
When it was in a state reactor the reactor wasn't being created until the state was getting used, due to us lazily defining states only when data is added to them. I had to move it to a system to get it to be active immediately on entering a scene

Reply:
gotcha ok, i would have thought the GLTF loader would initialize it before it was needed, but this makes sense